tlv/internal/gen/gen_tlv_types.go

```diff
@@ -8,8 +8,9 @@ import (
 )
 
 const (
-	numberOfTypes     = 100
-	defaultOutputFile = "tlv_types_generated.go"
+	numberOfTypes     uint32 = 100
+	customTypeStart   uint32 = 65536
```

Comment:
Maybe we should just mention here where the reasoning for `65536` can be found. Maybe something like:
```
	// customTypeStart is the inclusive start of the custom tlv type range as defined in BOLT 01.
```